File: mobile/baseplay_app/lib/features/documents/domain/models/document_type_model.dart
class DocumentTypeModel {
  final int id;
  final String name;
  final bool requiresExpiration;
  final int? defaultValidDays;

  const DocumentTypeModel({
    required this.id,
    required this.name,
    required this.requiresExpiration,
    required this.defaultValidDays,
  });

  factory DocumentTypeModel.fromJson(Map<String, dynamic> json) {
    return DocumentTypeModel(
      id: int.tryParse('${json['id'] ?? 0}') ?? 0,
      name: '${json['name'] ?? ''}',
      requiresExpiration: '${json['requires_expiration'] ?? '0'}' == '1',
      defaultValidDays: int.tryParse('${json['default_valid_days'] ?? ''}'),
    );
  }
}

File: app/Models/DocumentTypeModel.php
<?php

namespace App\Models;

use CodeIgniter\Model;

class DocumentTypeModel extends Model
{
    protected $table = 'document_types';
    protected $primaryKey = 'id';
    protected $returnType = 'array';

    protected $allowedFields = [
        'name',
        'requires_expiration',
        'default_valid_days',
        'status',
        'created_at',
        'updated_at',
    ];
}